WalletWasabi.Tests/UnitTests/WabiSabi/Models/MultipartyTransactionTests.cs

```diff
@@ -0,0 +1,327 @@
+using NBitcoin;
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using WalletWasabi.Helpers;
+using WalletWasabi.WabiSabi.Backend.Models;
+using WalletWasabi.WabiSabi.Models;
+using WalletWasabi.WabiSabi.Models.MultipartyTransaction;
+using WalletWasabi.Tests.Helpers;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Models
+{
+	public class MultipartyTransactionTests
+	{
+		private static MoneyRange DefaultAllowedAmounts = new (Money.Zero, Money.Coins(1));
+		private static Parameters DefaultParameters = new (FeeRate.Zero, DefaultAllowedAmounts, DefaultAllowedAmounts, Network.Main);
+
+		private static void ThrowsProtocolException(WabiSabiProtocolErrorCode expectedError, Action action) =>
+			Assert.Equal(expectedError, Assert.Throws<WabiSabiProtocolException>(action).ErrorCode);
+
+		[Fact]
+		public void TwoPartyNoFees()
+		{
+			using Key key1 = new();
+			using Key key2 = new();
+
+			var alice1 = WabiSabiFactory.CreateAlice(key: key1);
+			var alice2 = WabiSabiFactory.CreateAlice(key: key2);
+
+			var alice1Coin = alice1.Coins.First();
+			var alice2Coin = alice2.Coins.First();
+
+			var state = new Construction(DefaultParameters);
+
+			Assert.Empty(state.Inputs);
+			Assert.Empty(state.Outputs);
+
+			var oneInput = state.AddInput(alice1Coin);
+
+			Assert.Single(oneInput.Inputs);
+			Assert.Empty(oneInput.Outputs);
+
+			// Previous state should be unmodified
+			Assert.Empty(state.Inputs);
+			Assert.Empty(state.Outputs);
+
+			var differentInput = state.AddInput(alice2Coin);
+
+			Assert.Single(differentInput.Inputs);
+			Assert.Empty(differentInput.Outputs);
+			Assert.NotEqual(oneInput.Inputs, differentInput.Inputs);
+			Assert.Equal(oneInput.Outputs, differentInput.Outputs);
+
+			var twoInputs = oneInput.AddInput(alice2Coin);
+
+			Assert.Equal(2, twoInputs.Inputs.Count());
```

Comment:
```suggestion
			Assert.Equal(2, twoInputs.Inputs.Count);
```

This fixes a [CA1829](https://docs.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca1829) warning for me. L64, L72 and L181 generate CA1829s for me too.

Reply:
applying, I will force push after amending with the other lines so leaving unresolved for now